middlewares/validators/progresses.js

```diff
@@ -70,6 +70,25 @@ const validateGetProgressRecordsQuery = async (req, res, next) => {
         .messages({
           "string.base": "orderBy must be a string",
         }),
+      dev: joi.boolean().optional().messages({
+        "any.only": "dev field is restricted to either 'true' or 'false'",
+      }),
+      size: joi
+        .string()
+        .optional()
+        .pattern(/^[1-9]\d?$|^100$/)
+        .messages({
+          "string.empty": "size must contain value in range 1-100",
+          "string.pattern.base": "size must be in range 1-100",
+        }),
+      page: joi
+        .string()
+        .optional()
+        .pattern(/^0$|^[1-9]\d*$/)
```

Comment:
we have function for integer.

Reply:
I followed the same approach as in the user validators middleware, where this pattern is used for similar purposes. I just copied it from there to maintain consistency.

Reply:
I don't think we should follow the old approach. Joi provides validation for numbers and we should leverage its feature.

Reply:
okay fixing it

Reply:
resolved